### ISSUE TITLE
Cannot undo moving or copying synced clippings/folders after reloading Synced Clippings folder

### ISSUE THREAD

**Issue by aecreations** [aecreations/clippings#384]
Clippings 6.3.5

When the Synced Clippings folder is reloaded, the synced items are purged and reloaded from the sync file. This means that synced items are assigned new IDs in the IndexedDB database. But the undo and redo stacks still hold the old IDs of the synced items before the reload and purge, which causes the Undo and Redo actions to fail.

A temporary fix will be implemented for Clippings 6.3 - added error handling when the clipping or folder is invalid when editing.  But a better fix is needed.

**Comment by aecreations** [comment]
Not yet fully fixed. Still need to handle:

1. Creation of synced clippings and folders from Clippings Manager and the New Clipping dialog
2. Removing all source URLs in Clippings Manager
3. Upgrading from previous versions - all synced items need to be populated with static/synced IDs

Also, the static/synced IDs should be GUIDs, not random numbers. This change will ensure that synced items are unique across different instances of Clippings.

**Comment by aecreations** [comment]
If creating a new clipping from the New Clipping dialog, reloading the Synced Clippings folder from Clippings Manager should be disabled until the user dismisses the New Clipping dialog.

This will prevent errors due to the IDs of synced folders being changed when the sync data is reloaded and the IndexedDB data for them is refreshed.

**Comment by aecreations** [comment]
Remaining work to be deferred to after 6.4 beta 1:
1. Handling undo/redo of removing all source URLs in Clippings Manager
2. Temporarily disable reloading the Synced Clippings folder from Clippings Manager until the user dismisses the New Clipping dialog

**Comment by aecreations** [comment]
Still not fully fixed.  Steps to repro:
1. Select a clipping (or folder) in a sync folder and move it to the root folder, outside the Synced Clippings folder
2. Reload the Synced Clippings folder
3. Click Undo

Result: The move from step 1 is not undone.

**Comment by aecreations** [comment]
Despite these fixes, there are still caveats with undoing most Clippings Manager actions with synced items after reloading the Synced Clippings folder. Undo and redo may not work if the parent folder of a synced item was deleted or moved out of the Synced Clippings folder. Undoing or redoing removal of all source URLs will silently fail (no exceptions thrown) if the affected synced items were changed.

Trying to catch all these edge cases runs the risk of added complexity to Clippings Manager, as well as code bloat, making it harder to debug and maintain.  I have decided to leave it as it is for now, and will revisit if there is user demand.
